pkg/web/apps/apps_psql_repository.go

```diff
@@ -128,5 +130,154 @@ func (a *appsPostgreSQLRepository) GetAppByID(ID string) (*models.App, error) {
+
+	var device models.Device
+	if err := row.Scan(&device.ID, &device.VersionID, &device.AppID, &device.DeviceID, &device.DeviceType, &device.DeviceVersion); err != nil {
+		log.Error(err)
+		switch err {
+		case sql.ErrNoRows:
+			return nil, models.ErrNotFound
+		default:
+			return nil, models.ErrDatabaseError
+		}
+	}
+
+	return &device, nil
+}
+
+// GetVersionByAppIDAndVersion gets a version by its app ID and version number
+func (a *appsPostgreSQLRepository) GetVersionByAppIDAndVersion(appID string, versionNumber string) (*models.Version, error) {
+	version := models.Version{}
+	var disabledMessage sql.NullString
+	var lastLaunchedAt sql.NullString
+
+	sqlStatement := `
+	SELECT v.id,v.version,v.app_id, v.disabled, v.disabled_message, v.num_of_app_launches, v.last_launched_at
+	FROM version as v
+	WHERE v.app_id = $1 AND v.version = $2;`
+
+	err := a.db.QueryRow(sqlStatement, appID, versionNumber).Scan(&version.ID, &version.Version, &version.AppID, &version.Disabled, &disabledMessage, &version.NumOfAppLaunches, &lastLaunchedAt)
+
+	version.DisabledMessage = disabledMessage.String
+	version.LastLaunchedAt = lastLaunchedAt.String
+
+	if err != nil {
+		log.Error(err)
+		if err == sql.ErrNoRows {
+			return nil, models.ErrNotFound
+		}
+		return nil, models.ErrInternalServerError
+	}
+
+	return &version, nil
+}
+
+// GetDeviceByDeviceIDAndAppID returns a device by its device ID and app ID
+func (a *appsPostgreSQLRepository) GetDeviceByDeviceIDAndAppID(deviceID string, appID string) (*models.Device, error) {
+	device := models.Device{}
+
+	sqlStatement := `
+		SELECT d.id, d.version_id, d.app_id, d.device_id, d.device_type, d.device_version
+		FROM device as d
+		WHERE d.device_id = $1 AND d.app_id = $2;`
+
+	if err := a.db.QueryRow(sqlStatement, deviceID, appID).
+		Scan(&device.ID, &device.VersionID, &device.AppID, &device.DeviceID, &device.DeviceType, &device.DeviceVersion); err != nil {
+
+		log.Error(err)
+		if err == sql.ErrNoRows {
+			return nil, models.ErrNotFound
+		}
+		return nil, models.ErrInternalServerError
+	}
+
+	return &device, nil
+}
+
+// GetDeviceByVersionAndAppID returns a device by its version number and app ID
+func (a *appsPostgreSQLRepository) GetDeviceByVersionAndAppID(version string, appID string) (*models.Device, error) {
+	device := models.Device{}
+
+	sqlStatement := `
+		SELECT d.id, d.version_id, d.app_id, d.device_id, d.device_type, d.device_version
+		FROM device as d
+		WHERE d.app_id = $1 AND d.device_version = $2;`
+
+	if err := a.db.QueryRow(sqlStatement, appID, version).
+		Scan(&device.ID, &device.VersionID, &device.AppID, &device.DeviceID, &device.DeviceType, &device.DeviceVersion); err != nil {
+
+		log.Error(err)
+		if err == sql.ErrNoRows {
+			return nil, models.ErrNotFound
+		}
+		return nil, models.ErrInternalServerError
+	}
+
+	return &device, nil
+}
+
+// GetAppByID retrieves an app by id from the database
+func (a *appsPostgreSQLRepository) GetAppByAppID(appID string) (*models.App, error) {
```

Comment:
Small change, I would move this function to below the GetAppByID function.

Reply:
I believe that it is duplicated. Please, @craicoverflow could you rebase your PR with the master and check if it is already implemented?

Reply:
I got why it is here now 👍